helper/consts/consts.go

```diff
@@ -4,4 +4,6 @@ const (
 	// ExpirationRestoreWorkerCount specifies the numer of workers to use while
 	// restoring leases into the expiration manager
 	ExpirationRestoreWorkerCount = 64
+
+	VaultKVCLIClientHeader = "X-Vault-Kv-Client"
```

Comment:
Maybe `X-Vault-KV-Client` instead? Not sure what's the convention for headers in general in terms of casing.

Reply:
Yeah the convention is only the first letter of each work is capitalized, and in fact the http lib will change it to this format regardless of what it's set to.

Reply:
Related to #4172, we can change this entry to be part of a whitelist `[]string`, which then gets appended to `passthroughRequestHeaders` in https://github.com/hashicorp/vault/pull/4172/files#diff-00cc39793f8052ea5a2a66799942ca1fR475.

The other option would be to include this as a default value in `MountConfig.PassthroughRequestHeaders` during secret/auth mounting, but the danger there is that this now becomes a tunable value that can be accidentally overwritten which is likely to cause issues on the VKV backend if they forget to include as part of the list when mount tuning.

I'd prefer the former; have this be a part of a separate internal whitelist that we append to in `routeCommon()`.